src/libraries/System.Private.CoreLib/src/System/IndexOfAnyAsciiSearcher.cs

```diff
@@ -0,0 +1,333 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using System.Runtime.Intrinsics.X86;
+
+namespace System
+{
+    internal static class IndexOfAnyAsciiSearcher
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsSupportedAndProfitable(int searchSpaceLength) =>
+            (Ssse3.IsSupported || AdvSimd.Arm64.IsSupported) &&
+            searchSpaceLength >= Vector128<short>.Count;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe bool TryComputeBitmap(ReadOnlySpan<char> values, byte* bitmap)
+        {
+            byte* bitmapLocal = bitmap;
+
+            foreach (char c in values)
+            {
+                if (c > 127)
+                {
+                    return false;
+                }
+
+                int highNibble = c >> 4;
+                int lowNibble = c & 0xF;
+
+                bitmapLocal[lowNibble] |= (byte)(1 << highNibble);
```

Comment:
```suggestion
                bitmapLocal[(uint)lowNibble] |= (byte)(1 << highNibble);
```
To get rid of the sign-extending move.